lib/ansible/galaxy/collection.py

```diff
@@ -604,8 +605,19 @@ def download_collections(collections, output_path, apis, validate_certs, no_deps
                 requirements.append({'name': collection_filename, 'version': requirement.latest_version})
 
                 display.display("Downloading collection '%s' to '%s'" % (name, dest_path))
-                b_temp_download_path = requirement.download(b_temp_path)
-                shutil.move(b_temp_download_path, to_bytes(dest_path, errors='surrogate_or_strict'))
+
+                if requirement.api is None and requirement.b_path and os.path.isfile(requirement.b_path):
+                    shutil.copy(requirement.b_path, to_bytes(dest_path, errors='surrogate_or_strict'))
+                elif requirement.api is None and requirement.b_path:
+                    temp_path = to_text(b_temp_path, errors='surrogate_or_string')
+                    scm_build_path = os.path.join(temp_path, 'tmp_build')
+                    os.makedirs(scm_build_path, mode=0o0755)
```

Comment:
`os.makedirs` fails if the final dir already exists and the kwargs `exist_ok` was only added in Python 3. `scm_build_path` doesn't have any unique values in it and would cause a conflict if more than 1 download hit this path.